### PR TITLE
[#931]Code view: Default sorting in descending order

### LINKED ISSUE #931: Code view: default sorting should be LoC in descending order
Current: the default order is ascending
Suggestion: showing bigger changes on top make more sense as they are the more interesting ones

image

### PR DESCRIPTION
**Pull request by Tejas2805**
Fixes #931 

```
Currently, in Code View, the default order is ascending.

Let's make the default order descending as it is more meaningful to have
bigger changes on top as they are the more interesting changes.
```

### CODE REVIEW

**Comment by damithc** [comment]
Let's try to merge this soon as it can be useful for CS2103 PE

**Comment by damithc** [comment]
If the files are sorted in descending order, the `reverse` tick box must be ticked by default?

**Comment by jylee-git** [comment]
image

The checkbox should be checked by default if we are implementing this PR; else it won't make sense here.

**Comment by Tejas2805** [comment]
> If the files are sorted in descending order, the `reverse` tick box must be ticked by default?

I was thinking the same but then I thought its better to not have it ticked by default as it will be a bit weird. 

We are thinking from this point of view cause we know that ascending was the default before and descending is reverse. But for someone who is using it for the first time, descending will be the default and ascending will become reverse. So I think it's okay to not have it ticked from before.

@damithc Let me know if you still think that the box should be ticked by default is how we should proceed ahead. I will do the change.

**Comment by damithc** [comment]
> We are thinking from this point of view cause we know that ascending was the default before and descending is reverse. But for someone who is using it for the first time, descending will be the default and ascending will become reverse. So I think it's okay to not have it ticked from before.

That is a good point.  I think the better solution is to change it to a drop-down with options `Ascending` and `Descending`?

**Comment by Tejas2805** [comment]
> > We are thinking from this point of view cause we know that ascending was the default before and descending is reverse. But for someone who is using it for the first time, descending will be the default and ascending will become reverse. So I think it's okay to not have it ticked from before.
> 
> That is a good point. I think the better solution is to change it to a drop-down with options `Ascending` and `Descending`?

Can do this, with the default being `Descending`.

**Comment by Tejas2805** [comment]
@damithc Done.

**Comment by damithc** [comment]
> @damithc Done.

I'm OK with the UI. Can proceed to dev reviews.

File: frontend/src/static/js/v_authorship.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ window.vAuthorship = {
       totalLineCount: '',
       totalBlankLineCount: '',
       filesSortType: 'lineOfCode',
-      toReverseSortFiles: false,
+      toReverseSortFiles: 'descending',
```
Comment (jylee-git) [inline frontend/src/static/js/v_authorship.js]: Maybe rename this variable name to `sortByOrder`.

Comment (Tejas2805) [reply frontend/src/static/js/v_authorship.js]: Yeah makes sense. Will do the needful.

**Comment by Tejas2805** [comment]
> Let's try to merge this soon as it can be useful for CS2103 PE

@eugenepeh Could you please review the PR? The Professor thinks this might be useful for the CS2103 PE which is on Friday, so if all looks okay, we can also merge.

File: frontend/src/static/js/v_authorship.js [hunk 2]
```diff
@@ -210,7 +211,7 @@ window.vAuthorship = {
     },
 
     sortFiles() {
-      this.sortingFunction = (a, b) => (this.toReverseSortFiles ? -1 : 1)
+      this.sortingFunction = (a, b) => (this.sortByOrder === 'descending' ? -1 : 1)
```
Comment (eugenepeh) [inline frontend/src/static/js/v_authorship.js]: all these seems like unnecessary change when u can just bind the value true or false to `toReverseSortFiles`;
changing a binary value option to string compare is opening up spaces for bugs like typos/errors

Comment (Tejas2805) [reply frontend/src/static/js/v_authorship.js]: The `value` property of an HTML `option` element can only be a string. 

So if we want to do something like this - I will have to change the dropdown back to a checkbox. Let me know how you want me to proceed.

Comment (eugenepeh) [reply frontend/src/static/js/v_authorship.js]: Do look up `v-bind:`, take note of the syntax that we can utilize.

Comment (Tejas2805) [reply frontend/src/static/js/v_authorship.js]: Just checked. I have accordingly made the changes. Thanks for guiding

File: frontend/src/static/js/v_authorship.js [hunk 3]
```diff
@@ -199,6 +199,7 @@ window.vAuthorship = {
       this.isLoaded = true;
 
       this.activeFilesCount = this.selectedFiles.length;
+      this.sortFiles();
```
Comment (eugenepeh) [inline frontend/src/static/js/v_authorship.js]: this should be able to be removed since the value of toReverseSortFiles is watched

Comment (Tejas2805) [reply frontend/src/static/js/v_authorship.js]: We cannot remove it because even if `toReverseSortFiles` is watched, there is nothing to activate it. So, the default sorting is still ascending order, which defeats the purpose of the PR. Please refer to the screenshot below.

Screen Shot 2019-11-20 at 8 12 55 PM

Comment (eugenepeh) [reply frontend/src/static/js/v_authorship.js]: @Tejas2805 I am going to let you figure this out yourself so that maybe you will understand how vue works better this way.

Think about why there isn't a need to add the function there in the first place, yet the files will came out in an arranged order.

Comment (Tejas2805) [reply frontend/src/static/js/v_authorship.js]: I know that cause `toReverseSortFiles` is watched, whenever we change the sorting order, it will automatically cause the files to sort themselves, based on the value in `toReverseSortFiles` since its being `watched` but won't we have to activate it the first time to sort the files in required order? I tried running the program by removing `sortFiles()`  and the files came in ascending order with `order` as Descending.
So I am confused as to whether you are saying that we shouldn't need `sortFiles()` at all, or should I call the function somewhere else.

Comment (eugenepeh) [reply frontend/src/static/js/v_authorship.js]: >So I am confused as to whether you are saying that we shouldn't need sortFiles() at all, or should I call the function somewhere else.

Yes, you dont need sortFiles at all, you just need to change some code somewhere to make the default descending.

Comment (Tejas2805) [reply frontend/src/static/js/v_authorship.js]: @eugenepeh I have made the change. Wished to understand something- Is this way preferred cause this reduces the call stack?

Comment (fzdy1914) [reply frontend/src/static/js/v_authorship.js]: The original implementation will increase one additional unnecessary event loop of Vue, which is quite costly if there are multiple files.
